2333/2/libs/rapper.rb

```diff
@@ -0,0 +1,36 @@
+require 'russian_obscenity'
+
+# class Rapper
+class Rapper
+  attr_reader :name, :battles
+
+  def initialize(name, battles)
+    @name = name
+    @battles = battles
+  end
+
+  def stats
+    [@name, @battles.count, all_mats, mats_on_battle, words_on_round]
+  end
+
+  def all_mats
+    mats = 0
+    RussianObscenity.dictionary = [:default, './my_dictionary.yml']
+    RussianObscenity.find(@battles.join(' ')).each do |word|
```

Comment:
Very good catch that `.find` only returns unique values.

But I don't like that we create temporary variable `mats` here. Let's find all the unique bad words, map them to the number of times they appear in battles and then sum those numbers.